lib/shopify_cli/services/app/create/rails_service.rb

```diff
@@ -173,7 +179,7 @@ def build(name, db)
 
             context.puts(context.message("core.app.create.rails.adding_shopify_gem"))
             File.open(File.join(context.root, "Gemfile"), "a") do |f|
-              f.puts "\ngem 'shopify_app', '>=17.0.3'"
+              f.puts "\ngem 'shopify_app', '>=18.1.0'"
```

Comment:
Another food-for-thought: Are we concerned that some future version wouldn't be backwards compatible? Maybe `~>18.1` would be better?